vitrina/datasets/views.py

```diff
@@ -1915,28 +1916,25 @@ def has_permission(self):
         return has_perm(self.request.user, Action.VIEW, self.object)
 
     def get_queryset(self):
-        return get_projects(self.request.user, self.object, order_value="-created")
+        return get_projects(self.request.user).filter(datasets=self.object)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["dataset"] = self.object
-        context["can_add_projects"] = has_perm(
-            self.request.user,
-            Action.UPDATE,
-            self.object,
-        )
         context["can_view_members"] = has_perm(
             self.request.user,
             Action.VIEW,
             Representative,
             self.object,
         )
-        if self.request.user.is_authenticated:
-            context["has_projects"] = get_projects(
-                self.request.user, self.object, check_existence=True, form_query=True
-            )
-        else:
-            context["has_projects"] = False
+        context["available_projects"] = get_projects_linkable_to_dataset(self.request.user).exclude(
+            datasets=self.object
+        )
+        context["removable_project_ids"] = (
+            get_projects_linkable_to_dataset(self.request.user)
+            .filter(datasets=self.object)
+            .values_list("pk", flat=True)
+        )
```

Comment:
If you love performance and ordering doesn't matter.

```suggestion
        context["removable_project_ids"] = set(
            get_projects_linkable_to_dataset(self.request.user)
            .filter(datasets=self.object)
            .values_list("pk", flat=True)
        )
```